akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala

```diff
@@ -275,7 +275,8 @@ object ClusterEvent {
       val newMembers = newGossip.members -- oldGossip.members
       val membersGroupedByAddress = List(newGossip.members, oldGossip.members).flatten.groupBy(_.uniqueAddress)
       val changedMembers = membersGroupedByAddress collect {
-        case (_, newMember :: oldMember :: Nil) if newMember.status != oldMember.status ⇒ newMember
+        case (_, newMember :: oldMember :: Nil) if newMember.status != oldMember.status || newMember.upNumber != oldMember.upNumber ⇒
+          newMember
       }
       val memberEvents = (newMembers ++ changedMembers) collect {
```

Comment:
@viktorklang yes, that was something I learned from that presentation (surprised me that they are not implemented in the same way). I have changed: https://github.com/akka/akka/commit/811e28595f24de89210ceb85684c93fdb75ce0cb